### PR TITLE
feat: upgrade to rules_js 2.0

### PR DESCRIPTION
**Pull request by gregmagolan**
---

### Type of change

- New feature or functionality (change which adds functionality)

**For changes visible to end-users**

- Breaking change (this change will force users to change their own code or config)
- Suggested release notes are provided below:

### Test plan

- Covered by existing test cases

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -6,15 +6,11 @@ module(
     compatibility_level = 1,
 )
 
-# Lower-bound (minimum) versions of direct dependencies
+# Lower-bounds (minimum) versions for direct runtime dependencies
+bazel_dep(name = "aspect_bazel_lib", version = "2.7.1")
```
Comment (jbedard) [inline MODULE.bazel:10]: Should we try leaving some lower and just do rules_js + rules_nodejs? This was 2.5 in the `archive_override` can we keep that?

File: rollup/private/rollup.bzl [hunk 2]
```diff
@@ -215,19 +215,15 @@ def _impl(ctx):
...
             sources = output_sources_depset,
-            # Since we're bundling, we don't propogate linked npm packages from dependencies since
-            # they are bundled and the dependencies are dropped. If a subset of linked npm
-            # dependencies are not bundled it is up the the user to re-specify these in `data` if
-            # they are runtime dependencies to progagate to binary rules or `srcs` if they are to be
-            # propagated to downstream build targets.
-            transitive_npm_linked_package_files = npm_linked_packages.direct_files,
-            transitive_npm_linked_packages = npm_linked_packages.direct,
-            # Since we're bundling, we don't propogate any transitive sources from dependencies
+            types = depset(),  # rollup does not emit types directly
```
Comment (jbedard) [inline rollup/private/rollup.bzl:220]: Are these not optional and can be left unset?

File: MODULE.bazel (hunk 1 above)
Comment (gregmagolan) [reply MODULE.bazel:10]: Minimum has to be bumped to 2.7.1 to pickup the latest fixes for tar extract and for compat

File: rollup/private/rollup.bzl (hunk 2 above)
Comment (gregmagolan) [reply rollup/private/rollup.bzl:220]: They are optional and can be left unset but I added them for completeness so there is no ambiguity. Commented on why `types` is an empty depset as well.

**Review by alexeagle — APPROVED** [review]

**Pull request merged**